### ISSUE TITLE
Cash App

### ISSUE THREAD

**Issue by bugQ** [PrivSec-dev/banking-apps-compat-report#453]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

Cash App

### Link to app

https://play.google.com/store/apps/details?id=com.squareup.cash

### App version

4.49.0

### Country of the app

United States

### Build Number

2024052100

### Device list

Pixel 8

### Profile app tested in

Secondary profile(s)

### Google Play installed?

Installed

### Where did you install this app from?

Google Play Store

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [X] Allowed
- [ ] Blocked

### Exploit protection compatibility mode

- [X] Enabled
- [X] Disabled

### Memory tagging extension (MTE)

- [ ] Enabled
- [X] Disabled
- [ ] N/A

### Stock OS compatibility

- [X] Works
- [ ] Does not work
- [ ] Not tested

### NFC payments

- [ ] Works
- [ ] Works but requires another service
- [ ] Does not work
- [ ] N/A (Not supported by app)
- [X] Not tested

### Description of the app's functionality

Attempts to log into an existing account are met with "We couldn't verify this account belongs to you", regardless of login method, including all of the methods available under "I need help logging in".

In situations like this, Cash App support recommends creating a new account and verifying ownership over live chat via the new account.

However, attempts to create a new account are met with "Cash App doesn't support the use of emulators or other software programs."

### Are there any extra notes you think users should know about?

I have not exhausted all available combinations of OS settings yet. I may update this issue once I have.

**Comment by spring-onion** [comment]
Play integrity might be knocking on the door here. Mind capturing some logs?

**Comment by bugQ** [comment]
Updated issue to reflect that I have now tested both with and without Exploit Protection Compatibility Mode.

> Play integrity might be knocking on the door here. Mind capturing some logs?

Sure, I can try, hopefully soon.

**Comment by spring-onion** [comment]
A user reported it works if you revoke network access to all 3 Google Play components: Google Play services, Google Play Store, Google Services Framework.

**Comment by bugQ** [comment]
> A user reported it works if you revoke network access to all 3 Google Play components: Google Play services, Google Play Store, Google Services Framework.

Interesting, glad somebody got it to work, tho I don't seem to be able to replicate this.

For diagnosing this issue, would logs from just the app be enough, or do you need them from Play Services as well, or the whole system ?

**Comment by spring-onion** [comment]
Try clearing the app storage too.

For logs, the app ones alone will do for now.

**Comment by ThisNekoGuy** [comment]
Ran into this issue when I migrated to GrapheneOS the other day; here's an app log:

[Cash App log 14f72578fc80.txt](https://github.com/user-attachments/files/15958091/Cash.App.log.14f72578fc80.txt)

<details><summary>Screenshot:</summary>

![Screenshot_20240621-214302](https://github.com/PrivSec-dev/banking-apps-compat-report/assets/28468662/c48447fb-1736-4ef8-bdd8-0db1f11be58c)
</details>

**Comment by spring-onion** [comment]
@ThisNekoGuy Welcome! Have you tried the troubleshooting procedure I noted further up?

**Comment by ThisNekoGuy** [comment]
I saw it, but isn't revoking network access for those counterproductive (required) for literally every other app besides this one?
I was hesitant for that "solution" when I read it :/

**Comment by spring-onion** [comment]
@ThisNekoGuy It's only a temporary measure, just to check. If it turns out it works you could shove it into a secondary profile to not disturb the regular flow in your primary one.

**Comment by ThisNekoGuy** [comment]
I just tried it and it didn't work. :/
Not sure how that other person managed that by doing it.

**Comment by randomwithnoname** [comment]
If it's not compatible, shouldn't it be c̶r̶o̶s̶s̶e̶d̶ ̶o̶u̶t̶ on the website?

**Comment by spring-onion** [comment]
@randomwithnoname Thank you for bringing this to my attention, I'll fix it right away.

**Comment by spring-onion** [comment]
@bugQ @ThisNekoGuy @randomwithnoname 

I saw some promising feedback on twitter, please retest with the latest release (2025012700). Let me know if you end up playing around with the new play integrity toggle.

**Comment by ThisNekoGuy** [comment]
Where's the toggle?

**Comment by spring-onion** [comment]
@ThisNekoGuy It only shows up if the app actually used play integrity, otherwise the entry remains hidden. It's done this way to reduce unnecessary clutter.

**Comment by ThisNekoGuy** [comment]
Then perhaps it isn't using Play Integrity to do this? 🤔

**Comment by spring-onion** [comment]
Yeah, and even more mysterious is why it seemingly works for some.

**Comment by ThisNekoGuy** [comment]
@spring-onion Well, I updated Cash App and force closed it as well as clearing it's data... And it did set off the Play Integrity API notification this time. But it still didn't work. :/

![Image](https://github.com/user-attachments/assets/ffc5faaa-75ca-46c4-87ff-2da7a7fb3cf5)

I sent an email to squareup (`support@squareup.com`) about it, but I got an automatic reply that denies that they "can" reply to emails and that their "support center" be used
(which is basically just a goose chase for "use the support mechanism in the app")

**Comment by spring-onion** [comment]
@ThisNekoGuy And you tried blocking further attempts too I imagine?

**Comment by ThisNekoGuy** [comment]
No, I didn't... I assumed I wasn't supposed to if I intended it to work?

**Comment by spring-onion** [comment]
Some apps messed up the implementation and let you pass if they get nothing returned, so try it out.

**Comment by ThisNekoGuy** [comment]
Doing that got me past the ban, but the following set up to log in and "verify" my identity fails regardless of the options I choose for some reason. The options were to either verify with my card number or let it read my contacts to check for a match: but neither actually succeeded. :/

**Comment by NarwhalPrince** [comment]
I get the same behavior as before when blocking the Play Integrity request. Login goes smoothly until I enter my PIN (after verification code) and it just boots me back to the login saying "We couldn't verify this account belongs to you."

**Comment by ThisNekoGuy** [comment]
I also get that error message, but not until after it asked me for additional verification :/

**Comment by spring-onion** [comment]
@ThisNekoGuy @NarwhalPrince I saw some promising reports today, mind trying again?

**Comment by spigot** [comment]
Cleared cache and storage, login via email address worked for me in a secondary profile with no google services installed.
Native code debugging disabled
Exploit protection compat mode disabled

**Comment by ThisNekoGuy** [comment]
I also had to login with my email; it claimed I've "tried to register too many times" with my phone number

That aside, it did work (although I had the block still enabled from my last attempts) and the other settings left to their defaults

**Comment by bugQ** [comment]
I can confirm that it works with default runtime settings. Amazing!

> I saw some promising reports today, mind trying again?

@spring-onion May I ask where these reports originated? Did they coincide with the recent GrapheneOS release (2025050500), or did Square change something?

**Comment by spring-onion** [comment]
> [@spring-onion](https://github.com/spring-onion) May I ask where these reports originated? Did they coincide with a recent GrapheneOS release (perhaps [2025050300](https://grapheneos.org/releases#2025050300)), or did Block, Inc change something?

From the chat rooms just before I posted here. Someone randomly picked it up again, it worked for them and then another user confirmed the same. I'm not aware of any recent changes that would have affected app compatibility.

**Comment by ToxicSmurf** [comment]
I'm confused. Is it confirmed that the CashApp android app works on GrapheneOS?

**Comment by ThisNekoGuy** [comment]
I've been using it on GrapheneOS for a while, so pretty much

> I also had to login with my email; it claimed I've "tried to register too many times" with my phone number
> 
> That aside, it did work (although I had the block still enabled from my last attempts)
> 
> Hardened Memory: Enabled (default) Memory Tagging: Disabled (default) Native Code Debugging: Allowed (default) Enable Webview JIT: Enabled (default) Dynamic Code Loading (both): Enabled (default)
> 
> **Edit:** Allowing the API calls again let's the phone number features work (assigning a phone number to accounts, etc)
> 
> Although, it arbitrarily denies me permission to even try applying for the physical Cash App but I remember that being an issue for me even before I started using GrapheneOS (and not that their untenable support ever helped me with that either)

You'll have to allow the initial Integrity API attempt but after you login you can turn it back off

As for applying for the Cash App card (separate thing), they fixed the troublesome sign up procedure they used to have (it wouldn't work on GrapheneOS and their old support system was horrific) but the new backend approval process does work now also
